src/solver/td3.ml

```diff
@@ -395,90 +438,62 @@ module Base =
             )
           );
           if tracing then trace "sol2" "eval adding wpoint %a from %a" S.Var.pretty_trace y S.Var.pretty_trace x;
-          HM.replace wpoint y ();
+          mark_wpoint y default_widen_gas;
         );
         let tmp = simple_solve l x y in
         if HM.mem rho y then add_infl y x;
         if tracing then trace "sol2" "eval %a ## %a -> %a" S.Var.pretty_trace x S.Var.pretty_trace y S.Dom.pretty tmp;
         tmp
       and side ?x y d = (* side from x to y; only to variables y w/o rhs; x only used for trace *)
-        if tracing then trace "sol2" "side to %a (wpx: %b) from %a ## value: %a" S.Var.pretty_trace y (HM.mem wpoint y) (Pretty.docOpt (S.Var.pretty_trace ())) x S.Dom.pretty d;
+        if tracing then trace "sol2" "side to %a (wpx: %s) from %a ## value: %a" S.Var.pretty_trace y (format_wpoint y) (Pretty.docOpt (S.Var.pretty_trace ())) x S.Dom.pretty d;
         if Hooks.system y <> None then (
           Logs.warn "side-effect to unknown w/ rhs: %a, contrib: %a" S.Var.pretty_trace y S.Dom.pretty d;
         );
         assert (Hooks.system y = None);
         init y;
-        (match x with None -> () | Some x -> if side_widen = "unstable_self" then add_infl x y);
+
+        WPS.notify_side wps_data x y;
+
         let widen a b =
           if M.tracing then M.traceli "sol2" "side widen %a %a" S.Dom.pretty a S.Dom.pretty b;
           let r = S.Dom.widen a (S.Dom.join a b) in
           if M.tracing then M.traceu "sol2" "-> %a" S.Dom.pretty r;
           r
         in
         let old_sides = HM.find_default sides y VS.empty in
-        let op a b = match side_widen with
-          | "sides-local" when not (S.Dom.leq b a) -> (
```

Comment:
This `leq` check is not present in `SidesLocal.veto_widen`.

Reply:
That is true, but if b is less or equal to a, widening should have no effect, right?